src/main/java/org/spongepowered/api/conversation/Conversation.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.conversation;
+
+import org.spongepowered.api.conversation.QuestionResult.QuestionResultType;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.event.cause.conversation.ConversationEndType;
+import org.spongepowered.api.plugin.PluginContainer;
+import org.spongepowered.api.text.Text;
+
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents a created conversation with one or more {@link Conversant}s.
+ */
+public interface Conversation {
+
+    /**
+     * Gets who started the conversation, if present.
+     *
+     * @return The {@link PluginContainer} who started this conversation
+     */
+    PluginContainer getCreator();
+
+    /**
+     * Gets a data container where you can store all relevant information you
+     * gather from each question's {@link AnswerHandler}.
+     *
+     * @return The overall context of this conversation
+     */
+    DataContainer getContext();
+
+    /**
+     * An ease of access method for the string which a {@link Conversant} can
+     * use to exit a conversation.
+     *
+     * @return The exit string
+     */
+    default String getExitString() {
```

Comment:
I used the raw message(String s) from the implementations though before even the events are called for the time I check this.  No reason to really turn a string into a piece of text when it will just be turned right back into a string. I suppose it also takes more memory to store a text in the archetype too, but that's pretty insignificant. 

I am not completely against changing this to Text, but I don't feel it's necessary.